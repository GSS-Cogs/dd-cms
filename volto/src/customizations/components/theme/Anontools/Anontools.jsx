/**
 * Anontools component.
 * @module components/theme/Anontools/Anontools
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import config from '@plone/volto/registry';
import { Button } from 'govuk-react-jsx';

/**
 * Anontools container class.
 * @class Anontools
 * @extends Component
 */
export class Anontools extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    token: PropTypes.string,
    content: PropTypes.shape({
      '@id': PropTypes.string,
    }),
  };

  /**
   * Default properties.
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    token: null,
    content: {
      '@id': null,
    },
  };

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { settings } = config;
    return (
      !this.props.token && (
        <Link
          aria-label="login"
          to={`/login${
            this.props.content
              ? `?return_url=${this.props.content['@id'].replace(
                  settings.apiPath,
                  '',
                )}`
              : ''
          }`}
          id="ddcms_login"
        >
          <Button className="govuk-!-margin-0">Log in</Button>
        </Link>
      )
    );
  }
}

export default connect((state) => ({
  token: state.userSession.token,
  content: state.content.data,
}))(Anontools);
