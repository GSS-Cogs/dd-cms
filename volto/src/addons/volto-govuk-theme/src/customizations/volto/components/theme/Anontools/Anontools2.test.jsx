import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import Anontools from './Anontools';

const mockStore = configureStore();

describe('Anontools', () => {
  it('should render an anontools component when a token is specified to logout', async (done) => {
    const store = mockStore({
      userSession: { token: '1234' },
      content: { data: { '@id': 'myid' } },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = await renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Anontools />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
    done();
  });
});
