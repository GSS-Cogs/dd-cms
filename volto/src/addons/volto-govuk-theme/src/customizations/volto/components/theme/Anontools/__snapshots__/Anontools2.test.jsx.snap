// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anontools should render an anontools component when a token is specified to logout 1`] = `
<a
  aria-label="logout"
  href="/logout?return_url=myid"
  id="ddcms_login"
  onClick={[Function]}
>
  <button
    className="govuk-button govuk-!-margin-0 "
    data-module="govuk-button"
  >
    Log out
  </button>
</a>
`;
